docs/data/tree-view/tree-item-customization/CheckboxSlot.js

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import { RichTreeView } from '@mui/x-tree-view/RichTreeView';
+import { TreeItem2 } from '@mui/x-tree-view/TreeItem2';
+import { MUI_X_PRODUCTS } from './products';
+
+function CustomCheckbox(props) {
+  return <input type="checkbox" {...props} />;
```

Comment:
When you click on the checkbox it toggles the expansion of the item
I think it's due to you not passing the ref to the DOM

Reply:
nice catch, thanks!